site/cds_rdm/inspire_harvester/writer.py

```diff
@@ -187,6 +189,19 @@ def _create_new_version(self, entry, record):
                 system_identity, new_version_draft.id
             )
 
+    def _add_community(self, draft):
+        """Add CERN Scientific Community to the draft."""
+        with db.session.begin_nested():
```

Comment:
could you explain why we need begin_nested here and only here? I kinda have a feeling it should be used in a broader context